### ISSUE TITLE
React v15.0

### ISSUE THREAD

**Issue by tkawachi** [Hacker0x01/react-datepicker#444]
React v15.0 was released.
http://facebook.github.io/react/blog/2016/04/07/react-v15.html

**Comment by martijnrusschen** [comment]
Will upgrade this weekend if no-one beats me.

**Comment by rafeememon** [comment]
We need https://github.com/souporserious/react-tether to upgrade before we can do this

**Comment by martijnrusschen** [comment]
Opened a PR for react-tether: https://github.com/souporserious/react-tether/pull/15

**Comment by joelzimmer** [comment]
Hey guys, when will this be merged with a new version?

**Comment by halloffame** [comment]
Looks like a new version of react-tether was released with the dependency change
v0.5.1
https://www.npmjs.com/package/react-tether

**Comment by pocketjoso** [comment]
Would be great to get this bumped!

**Comment by rafeememon** [comment]
Looks like [redbox-react](https://github.com/KeywordBrain/redbox-react) also needs an updated dependency.  Also could ditch this entirely; the console is good enough for error messages IMO

**Comment by joelzimmer** [comment]
We'd really appreciate a version bump since it would help us upgrade one of our projects to use React 15.

**Comment by halloffame** [comment]
I'm all for ditching dependencies if they are not really necessary.

**Comment by martijnrusschen** [comment]
Lets try to keep redbox-react in the stack is possible. It's a great tool when developing the examples site.

@halloffame It's just a devDependency. Doesn't hurt the component itself.

I'll try to make the dependency compatible with React 15.

**Comment by rafeememon** [comment]
> It's just a devDependency. Doesn't hurt the component itself.

TBH I think this makes it even more silly that it's blocking the upgrade.  :stuck_out_tongue:  Let's hope that your PR gets merged soon.

**Comment by martijnrusschen** [comment]
Agree lets get rid of redbox-react for now :)

**Comment by rafeememon** [comment]
Unfortunately there's yet another issue -- the `react-tether` library now only accepts strings (i.e. query selectors) instead of objects (i.e. DOM elements) for the `renderElementTo` prop.  This will need to be accounted for as well before we can upgrade.

**Comment by rafeememon** [comment]
`renderElementTo` fix: https://github.com/souporserious/react-tether/pull/19

**Comment by michaelzoidl** [comment]
Hey guys, is there some work-around to get the `react-datepicker` working?
In another project just removing `react v0.14` from the `peerDependencies` in the `package.json` worked, seems like this works not for this project.

**Comment by rafeememon** [comment]
We can't remove react from peerDependencies and we need to support our API, so until the fix for react-tether is merged and released we can't upgrade to v15.

**Comment by peXed** [comment]
Hello! @rafeememon how is the status on this? React-tether is now up-to-date with their dependencies again or is still something missing?

**Comment by rafeememon** [comment]
@peXed, waiting for https://github.com/souporserious/react-tether/pull/19 to merge

**Comment by rafeememon** [comment]
Since there isn't any movement on the react-tether PR, I think we should probably inline the `TetherComponent` into this repo and make the changes we need.  Hopefully I can take care of this tonight.

**Comment by orangecoding** [comment]
@rafeememon is there a plan when React 15 will finally be supported as I would love to use this within my project. Anything I can help with to speed things up?

**Comment by martijnrusschen** [comment]
We're almost there. #469 should remove the latest blocker.

**Comment by orangecoding** [comment]
k. If there's anything I can help with, just comment.

**Comment by rafeememon** [comment]
@martijnrusschen, I merged the final change in #451.  I think it's time to cut a release :beers:

**Comment by martijnrusschen** [comment]
I'll do a release tomorrow :)

**Comment by martijnrusschen** [comment]
Done: https://github.com/Hacker0x01/react-datepicker/releases/tag/v0.27.0

**Comment by orangecoding** [comment]
@martijnrusschen awesome mate. thanks!
